src/nvim/CMakeLists.txt

```diff
@@ -878,16 +878,6 @@ set(VIMDOC_FILES
   ${NVIM_RUNTIME_DIR}/doc/treesitter.txt
 )
 
-glob_wrapper(API_SOURCES ${PROJECT_SOURCE_DIR}/src/nvim/api/*.c)
```

Comment:
These changes _seem_ unrelated? (@dundargoc)

Reply:
They are just so `make doc` triggers properly.

Reply:
Eh, he just moved the code and added more dependent files to the doc gen. This is probably more correct as those files are likely used by `gen_vimdoc.py`. As for the relevancy to this PR, I', unsure.

Reply:
So unrelated :) Should at least be an independent commit then.